.github/workflow/lint.yml

```diff
@@ -0,0 +1,31 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.22.3'
+
+      - name: Install dependencies
+        run: go mod download
+
+      - name: Install GolangCI-Lint
+        run: go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.50.0
```

Comment:
You can use `@latest` to get the latest version of golangci-lint. This will allow you to always use the latest version of the linter without having to manually update the configuration.